cmd/cstor-pool-mgmt/controller/pool-controller/new_pool_controller.go

```diff
@@ -113,6 +116,11 @@ func NewCStorPoolController(
 			if IsDeletionFailedBefore(cStorPool) || IsErrorDuplicate(cStorPool) {
 				return
 			}
+			if cStorPool.Labels[string(apis.OpenEBSVersionKey)] != curVersion {
```

Comment:
IMO we should tackle upgrade separately.
Can we avoid putting upgrade code into business logic?

Reply:
How about setting the pool controller deployment to 0 replicas before proceeding with upgrade?
What all impacts does this have?

Reply:
Alternatively, we may use below label which is about reconcile process and might solve upgrade requirements as well:
```yaml
labels:
  reconcile.openebs.io/disable: true # this is false by default
```

Reply:
> How about setting the pool controller deployment to 0 replicas before proceeding with upgrade?
> What all impacts does this have?

If upgrade went smooth I think we won't have any problem with that approach (I need to test it).
but in unsuccessful scenario clearly we can see the risk